### PR TITLE
[portsorch]: add support to set mac-address learning attribute on bridge-port

### PR DESCRIPTION
**Pull request by shine4chen**
**What I did**
add support to set mac-address learning attribute on bridge-port

**Why I did it**
Mclag feature need this.  And it is common attribute for bridge-port.

**How I verified it**
compile and test on nephos lab

**Details if related**

### CODE REVIEW

**Comment by lguohan** [comment]
learning mode can also be supported in lag, but it looks like the code here does not consider that scenario.

**Comment by jianjundong** [comment]
> learning mode can also be supported in lag, but it looks like the code here does not consider that scenario.

Lag is also supported, please see the function PortsOrch::doLagTask(Consumer &consumer). Thinks.

**Comment by leoli-nps** [comment]
Learning mode has been supported in lag, but the corresponding vs-test requires lag-name-oid-map. [PR#885](https://github.com/Azure/sonic-swss/pull/885) will provide support for lag-name-oid-map, so the vs-test code can only be provided after its modification is completed.

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -1813,7 +1820,48 @@ void PortsOrch::doPortTask(Consumer &consumer)
...
+                            else
+                            {
+                                SWSS_LOG_ERROR("Failed to set port %s learn mode to %s", alias.c_str(), learn_mode.c_str());
+                                it++;
+                                continue;
+                            }
+                        }
+                        else
+                        {
+                            m_portSetLearnModeList[alias] = learn_mode;
```
Comment (stcheng) [inline orchagent/portsorch.cpp]: could you add this to the current `port` class and later check if the variable `learn_mode` has value or not?

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -2013,7 +2061,38 @@ void PortsOrch::doVlanMemberTask(Consumer &consumer)
             }
 
             if (addBridgePort(port) && addVlanMember(vlan, port, tagging_mode))
```
Comment (stcheng) [inline orchagent/portsorch.cpp]: could you add this portion of the code into `addBridgePort` function and set the attribute during the bridge port creation time?

File: orchagent/portsorch.cpp [hunk 3]
```diff
@@ -2767,7 +2923,7 @@ bool PortsOrch::addLag(string lag_alias)
 
     PortUpdate update = { lag, true };
     notify(SUBJECT_TYPE_PORT_CHANGE, static_cast<void *>(&update));
-
+
```
Comment (stcheng) [inline orchagent/portsorch.cpp]: remove

**Comment by leoli-nps** [comment]
The vstest has been added, please review. Thanks!

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (shine4chen) [reply orchagent/portsorch.cpp]: yes, we will move learn_mode to port class. @stcheng

File: orchagent/portsorch.cpp (hunk 2 above)
Comment (shine4chen) [reply orchagent/portsorch.cpp]: Good point. we will refine code per your suggestion.

**Comment by shine4chen** [comment]
@stcheng we has refined code according to your suggest. Pls review thanks.

File: orchagent/portsorch.cpp [hunk 4]
```diff
@@ -2221,6 +2264,42 @@ void PortsOrch::doLagTask(Consumer &consumer)
                         gIntfsOrch->setRouterIntfsMtu(l);
                     }
                 }
+
+                string learn_mode;
+                for (auto i : kfvFieldsValues(t))
+                {
+                     /* Get port fdb learn mode*/
+                    if (fvField(i) == "learn_mode")
```
[the remark this answers is not in the document]
Comment (shine4chen) [reply orchagent/portsorch.cpp]: @lisahnguyen  good point. We will revise it.

**Comment by shine4chen** [comment]
retest it please

**Comment by shine4chen** [comment]
retest this please

**Pull request merged**